homeassistant/components/sensor/gtfs.py

```diff
@@ -213,104 +278,255 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class GTFSDepartureSensor(Entity):
     """Implementation of an GTFS departures sensor."""
 
-    def __init__(self, pygtfs, name, origin, destination, offset):
+    def __init__(self, pygtfs, name, origin, destination, offset) -> None:
         """Initialize the sensor."""
         self._pygtfs = pygtfs
         self.origin = origin
         self.destination = destination
         self._offset = offset
         self._custom_name = name
+
+        self._available = None
         self._icon = ICON
         self._name = ''
         self._unit_of_measurement = 'min'
         self._state = None
         self._attributes = {}
+
+        self._agency = None
+        self._departure = None
+        self._destination = None
+        self._origin = None
+        self._route = None
+        self._route_type = None
+        self._trip = None
+
         self.lock = threading.Lock()
         self.update()
 
     @property
-    def name(self):
+    def name(self) -> str:
```

Comment:
Opportunistic type hinting (and all methods below). Further type hinting to be performed in upcoming PRs.